posts/2021-07-28-GATs-stabilization-push.md

```diff
@@ -9,56 +9,87 @@ team: the Traits Working Group <https://www.rust-lang.org/governance/teams/compi
 *crickets chirping*
 
 Wait...that's it?? Well, yes! I'll go into a bit of detail later in this blog post as to why this *is* a big deal. But, long story short, there have been a good amount of changes that have had to have been made to the compiler to get GATs to work. And, while there are still a few small remaining diagnostics issues, the feature is finally in a space that we feel comfortable making it no longer "incomplete".
 
-So, what does that mean? Well, all it *really* means is that when you use this feature on nightly, you'll no longer get the "`generic_associated_types` is incomplete" warning. However, the real reason this is a big deal: we want to stabilize this feature. But we need *your* help. We need you to test this feature; file issues for bugs or for potential diagnostics improvements.
-
-Without making promises that we aren't 100% sure we can keep, we have high hopes we can stabilize this feature within the next couple months. But, we want to make sure we aren't missing glaringly obvious bugs or flaws. And that's why we need *your* help.
+So, what does that mean? Well, all it *really* means is that when you use this feature on nightly, you'll no longer get the "`generic_associated_types` is incomplete" warning. However, the real reason this is a big deal: we want to stabilize this feature. But we need your help. We need you to test this feature, to file issues for any bugs you find or for potential diagnostic improvements. Also, we'd love for you to just tell us about some interesting patterns that GATs enable over on [Zulip]!
 
+Without making promises that we aren't 100% sure we can keep, we have high hopes we can stabilize this feature within the next couple months. But, we want to make sure we aren't missing glaringly obvious bugs or flaws. We want this to be a smooth stabilization.
 
 Okay. Phew. That's the main point of this post and the most exciting news. But as I said before, I think it's also reasonable for me to explain *what* this feature is, what you can do with it, and some of the background and how we got here.
 
 ## So what are GATs?
 
 **Note: this will only be a brief overview. The [RFC] contains many more details.**
 
-GATs (generic associated types) were originally proposed in [RFC 1598][RFC]. Put simply, they allow you to define type, lifetime, or const generics on associated types. If you're familiar with languages that have "higher-kinded types", then you could calls GATs *type constructors on traits*.
+GATs (generic associated types) were originally proposed in [RFC 1598][RFC]. Put simply, they allow you to define type, lifetime, or const generics on associated types. If you're familiar with languages that have "higher-kinded types", then you could call GATs *type constructors on traits*.
 
 To jump into an example of GATs, here is a popular use case: a `LendingIterator` (formerly known as a `StreamingIterator`):
 
 ```rust
 trait LendingIterator {
-    type Item<'a>;
+    type Item<'a> where Self: 'a;
```

Comment:
Hmm why is the where clause necessary?

Reply:
Annoying, right? The short version is that the impls need to know that people can only use "reasonable" `'a` lifetimes here. e.g., the example impl below uses `type Item<'a> = &'a [T]` or something, but that is only valid if `T: 'a`. So what's to stop someone from doing `Foo::Item<'static>`, which might not be legal? This where clause, that's what. It'd be nice if we could find a way to avoid folks having to type it, though!